onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -201,103 +208,138 @@ Status TreeEnsembleCommon<InputType, ThresholdType, OutputType>::Init(
   // filling nodes
 
   n_nodes_ = nodes_treeids.size();
-  nodes_.resize(nodes_treeids.size());
+  limit = static_cast<size_t>(n_nodes_);
+  InlinedVector<TreeNodeElementId> node_tree_ids;  
+  node_tree_ids.reserve(limit);
+  nodes_.clear();
+  nodes_.reserve(limit);
   roots_.clear();
-  std::unordered_map<TreeNodeElementId, TreeNodeElement<ThresholdType>*, TreeNodeElementId::hash_fn> idi;
+  absl::flat_hash_map<TreeNodeElementId, uint32_t, TreeNodeElementId::hash_fn> idi;
```

Comment:
>absl::flat_hash_map [](http://example.com/codeflow?start=2&length=19)

We can not use Abseil types directly because of the requirement to have no Abseil builds, and the hash function type is not exposed. I guess, we would need to revert.

Reply:
reverted